extensions/kotlin/src/main/java/io/opentelemetry/extension/kotlin/ContextElement.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.extension.kotlin;
+
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import kotlin.coroutines.CoroutineContext;
+import kotlin.jvm.functions.Function2;
+import kotlinx.coroutines.ThreadContextElement;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * {@link ThreadContextElement} for synchronizing a {@link Context} across coroutine suspension and
+ * resumption. Implemented in Java instead of Kotlin to allow usage in auto-instrumentation where
+ * there is an outstanding Kotlin bug preventing it https://youtrack.jetbrains.com/issue/KT-20869.
+ */
+class ContextElement implements ThreadContextElement<Scope> {
```

Comment:
I wonder if we should prefix this with `Kotlin` or `Coroutine` to make reading code usages super clear about which context we're referring to.

Reply:
Yup good call